### ISSUE TITLE
Pokemon Stadium 1 and 2 GB tower

### ISSUE THREAD

**Issue by AlexN64gamer** [project64/project64#2055]
Please, developers, would it be possible to make this feature of these games available running at 100% speed in future releases of the emulator? thanks a lot in advance

P.S: I have already tested both of them, my pc specs:  Intel I7 10700K, RTX 2080 ti gaming, 32Gb ram Corsair vengeance pro, motherboard MSI Tomahawk Z490,  Roms do load into the the GB tower screen but theyre run totally unplayable (like 5% speed)

**Comment by LegendOfDragoon** [comment]
Just use cpu interpreter. This is a duplicate of #1461 and #964

**Comment by AlexN64gamer** [comment]
Ill try

**Comment by AlexN64gamer** [comment]
![Captura33](https://user-images.githubusercontent.com/38527600/119824878-48ee4200-beee-11eb-9d11-d0dca20582d7.PNG)

Tested on Pokemon Stadium 1 lastest dev version, cpu core style: interpreter pokemon red, it doesnt even boot

**Comment by AlexN64gamer** [comment]
![Captura34](https://user-images.githubusercontent.com/38527600/119825752-33c5e300-beef-11eb-9320-147c65f24a93.PNG)

Tested on Pokemon Stadium 2 lastest dev version, cpu core style: interpreter, the rom boots but the stability in-game on the GB tower is far from 100%. lots of loading times and fps drops which makes this game mode unplayable

**Comment by LegendOfDragoon** [comment]
Are you overclocking pj64? Also what counter factor? The slowdowns sound unusual. I don't recall the game running that poorly tbh.

**Comment by AlexN64gamer** [comment]
nope, I dont know how to overclock the emulator, and counter factor is 2, btw my specs are in my first post, I dont think my pc is the problem, im thinking to upload a small footage to show everything properly

**Comment by AlexN64gamer** [comment]
The problem still exists, tried P64 3.0.,1 stable version today, with PK stadium 1 and pk red .gb and .sav (through Nrage transfer pak emulation) GB tower doesnt evey boot

**Comment by AlexN64gamer** [comment]
https://www.youtube.com/watch?v=P80oQ7q-l2A

**Comment by LegendOfDragoon** [comment]
There is no good reason to repeatedly mention that these games do not work. It is a known issue and no one else has attempted to fix it, so what do you expect??

**Comment by Squall-Leonhart** [comment]
use parallel rdp and disable 32bit engine

**Comment by RevoSucks** [comment]
I need this to work since I work on the decompilation for Pokemon Stadium; otherwise I have no way to debug/breakpoint and document the GB tower while decompiling its functionality.

**Comment by Squall-Leonhart** [comment]
> I need this to work since I work on the decompilation for Pokemon Stadium; otherwise I have no way to debug/breakpoint and document the GB tower while decompiling its functionality.

Stadium 2 GB Tower settings

Set VI Refresh to 3000
Disable "32bit Engine"
Use Parallel-RSP for PJ64.

![image](https://github.com/user-attachments/assets/5aaad88f-fbf4-442d-9fe4-4ebea39b95c3)

Gameboy Tower in stadium 1 never passes the bootrom.

**Comment by RevoSucks** [comment]
Yeah, but Stadium 1 is what I need, since that is the decomp that currently exists. We have not started Stadium 2 yet.

**Comment by Squall-Leonhart** [comment]
I can't figure out the origin of the boot loop thats occuring in the stadium 1 version of GBEMU, but i assume its due to whack core timing.

**Comment by RevoSucks** [comment]
Do you know the general "fragment" or virtual address range this may be occurring in? We can try to prioritize its decompilation if that would help you.

**Comment by Squall-Leonhart** [comment]
at 8011B7BB and later section in rdram, theres some memory it attempts to set repeatedly, which then sets a region in SP DMEM (which i believe is transferpak data)

The same operation latches and boots first time on GBEMU2

**Comment by RevoSucks** [comment]
Im having trouble finding this. 8011B7B8-8011B7BC in VRAM at the time of the boot ROM hanging in GB Tower (in stadium US 1.0) does not look like valid code. Do you know what ROM address this code is in? This area is in a "heap" so it doesnt cleanly map to our ROM map in the decomp. I cant trace it back to the original function.

**Comment by Squall-Leonhart** [comment]
hi @RevoSucks 

AIO pointed me towards a change in PJ64 that can make stadium 1 boot, it requires changing the SI DMA duration from 2304 (0x900) to 8192 (0x2000) which is hard coded 3x within https://github.com/project64/project64/blob/develop/Source/Project64-core/N64System/MemoryHandler/PifRamHandler.cpp

The affects of this SI Delay can be affected by counterfactor more or less (CF 1 needs a lower SI Delay for instance)

Making this change, you then need to set the games settings to

Core: Interpreter
VI Refresh 3000 (mainly for full speed in game)
SI Delay On (Defaults to off)

Plugin
RSP: Parallel-RSP

Mupen NX will also soon have working transfer pack support too, after i bugged m4xw about it failing to load properly, 
Gopher and Ares also support it.

Something interesting to note, if you pair a gb rom with n64 rom of different language regions, (ie, German cart with US/EU gb rom) the need for a drawn out SIA delay is negated by a different loading screen used.
